solana/indexer/processor.go

```diff
@@ -27,23 +28,65 @@ type Processor interface {
 }
 
 type DefaultProcessor struct {
-	rpcClient   RpcClient
-	pool        DbPool
-	config      config.Config
-	mintsFilter *[]string
+	rpcClient        RpcClient
+	pool             DbPool
+	config           config.Config
+	transactionCache *otter.Cache[solana.Signature, *rpc.GetTransactionResult]
 }
 
-func (p *DefaultProcessor) ProcessSignature(ctx context.Context, slot uint64, txSig solana.Signature, logger *zap.Logger) error {
-	txRes, err := p.rpcClient.GetTransaction(
-		ctx,
-		txSig,
-		&rpc.GetTransactionOpts{
-			Commitment:                     rpc.CommitmentConfirmed,
-			MaxSupportedTransactionVersion: &rpc.MaxSupportedTransactionVersion0,
-		},
-	)
+func NewDefaultProcessor(
+	rpcClient RpcClient,
+	pool DbPool,
+	config config.Config,
+) *DefaultProcessor {
+	cache, err := otter.MustBuilder[solana.Signature, *rpc.GetTransactionResult](50).
+		WithTTL(30 * time.Second).
+		CollectStats().
+		Build()
+
 	if err != nil {
-		return fmt.Errorf("failed to get transaction: %w", err)
+		panic(fmt.Errorf("failed to create transaction cache: %w", err))
+	}
+	return &DefaultProcessor{
+		rpcClient:        rpcClient,
+		pool:             pool,
+		config:           config,
+		transactionCache: &cache,
+	}
+}
+
+func (p *DefaultProcessor) ProcessSignature(ctx context.Context, slot uint64, txSig solana.Signature, logger *zap.Logger) error {
+	var txRes *rpc.GetTransactionResult
+
+	// Check if the transaction is in the cache
+	if p.transactionCache != nil {
+		if hit, ok := p.transactionCache.Get(txSig); ok {
```

Comment:
is `solana.Signature` an alias for string or a more complicated type?

Reply:
alias for `byte[64]`